salt/states/test.py

```diff
@@ -334,7 +334,7 @@ def _check_key_type(key_str, key_type=None):
     value = __salt__['pillar.get'](key_str, None)
     if value is None:
         return None
-    elif type(value) is not key_type and key_type is not None:
+    elif key_type is not None and type(value) is not key_type and not isinstance(value, key_type):
```

Comment:
In (at least) python 2, in certain situations the pillar data may be OrderedDicts instead of normal dicts, and the strings may be unicode instead of just strings.  In this case, the `dictionary` check_pillar would fail since `type({})!=OrderedDict`, and the same for unicode strings.

I only noticed this problem when loading pillar data from flat files. Using extpillar providers (as we typically do) results in normal `dict` and `str` objects.

Reply:
But they should always pass the isinstance check, which i would say would be good to remove the type(value) is not key_type, and just use the not isinstance.

Reply:
Yes, I'm fine with that.

Reply:
Updated with this change